huxley/www/js/stores/DelegateStore.js

```diff
@@ -1,32 +1,79 @@
 /**
- * Copyright (c) 2011-2015 Berkeley Model United Nations. All rights reserved.
+ * Copyright (c) 2011-2016 Berkeley Model United Nations. All rights reserved.
  * Use of this source code is governed by a BSD License (see LICENSE).
  */
 
 'use strict';
 
-var $ = require('jquery');
+var ActionConstants = require('constants/ActionConstants');
 var Dispatcher = require('dispatcher/Dispatcher');
 var ServerAPI = require('lib/ServerAPI');
 var {Store} = require('flux/utils');
 
 
 var _delegatePromises = {};
+var _delegates = []
 
 class DelegateStore extends Store {
   getDelegates(schoolID, callback) {
     if (!_delegatePromises[schoolID]) {
       _delegatePromises[schoolID] = ServerAPI.getDelegates(schoolID);
     }
     if (callback) {
-      _delegatePromises[schoolID].then(callback);
+      _delegatePromises[schoolID].then(function(value) {
+        _delegates = value;
+        callback(value);
+      });
     }
-    return _delegatePromises[schoolID];
+    return _delegates;
+  }
+
+  deleteDelegate(delegate) {
+    ServerAPI.deleteDelegate(delegate.id);
+    var index = _delegates.indexOf(delegate);
```

Comment:
Assuming that we'd only pass the `delegateID` into this function:
```js
_delegates = _delegates.filter(d => d.id !== delegateID);
```

This sets `_delegates` to be a new array instead of mutating the old one. In general, we want to avoid mutative behaviors like this, because we don't know who else has a reference to these objects.